nncf/experimental/common/pruning/operations.py

```diff
@@ -0,0 +1,698 @@
+        """
+        The function tries to decompose one element in the source array by the product of sequential elements in the
+        target array. If it can be decomposed it's considered "mapped". i.e. the source element corresponds to the list
+        of elements in the target array, end vice-versa, elements in the target array corresponds to the element in the
+        source array. By this correspondence, mappings of element's indexes from source to target and from target to
+        source are built.
+
+        :param source_array: array to get a single element for mapping to elements in the target array.
+        :param target_array: array to find element(s) that map(s) to a single element in the source array.
+        :param source_idx: index of the single element in the source array.
+        :param target_start_idx: index in the target array to start sequentially multiplying elements until reaching
+            the element from source array.
+        :param source_to_target_map: mapping of indexes of source elements to indexes of target elements.
+        :param target_to_source_map: mapping of indexes of target elements to indexes of source elements.
+        :return: a pair, when first element is True when the element from source array can be mapped to sequential
+            element(s) of target array, False - otherwise.
+            The second element of pair is the index of last mapped element in the target array.
+        """
+        res, last_target_index = cls._can_reach_number_by_multiply(
+            number_to_reach=source_array[source_idx],
+            array=target_array,
+            start_idx=start_target_idx
+        )
+        if not res:
+            return (res, last_target_index)
+        source_to_target_map[source_idx] = list(range(start_target_idx, last_target_index + 1))
+        for idx in range(start_target_idx, last_target_index + 1):
+            target_to_source_map[idx] = [source_idx]
+        return (res, last_target_index)
+
+    @classmethod
+    def parse_reshape(cls,
+                      reshape_attributes: ReshapeLayerAttributes) -> Tuple[DIMENSION_MAP, DIMENSION_MAP, ReshapeMode]:
+        """
+        Parse reshape semantic by mapping elements in the input shape to elements of output shape.
+
+        :param reshape_attributes: attributes that contains input and output shape.
+        :return: return two mappings (from input to output, and vice versa) and reshape mode.
+        """
+        input_shape_not_cut = reshape_attributes.input_shape
+        output_shape_not_cut = reshape_attributes.output_shape
+
+        in_indexes_not_cut_map = [i for i, dim in enumerate(input_shape_not_cut) if dim != 1]
+        out_indexes_not_cut_map = [i for i, dim in enumerate(output_shape_not_cut) if dim != 1]
+
+        input_shape=list(filter(lambda x: x != 1, input_shape_not_cut))
+        output_shape=list(filter(lambda x: x != 1, output_shape_not_cut))
```

Comment:
We can still retrieve `input_shape_not_cut` and `output_shape_not_cut` from `reshape_attributes`

Reply:
Let's not be perfectionists :) 
Moreover, the proposed change is harmful because the responsibility for deleting 1's will be spread across two functions.
Currently, `_convert_to_not_cut` doesn't know about deleting 1's, it just does local thing.